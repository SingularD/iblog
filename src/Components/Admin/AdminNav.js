import React from 'react'

class AdminNav extends React.Component{
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div className="admin-nav-container col-2 row justify-content-center bg-light">
        <div className="col-10">
          <h2>后台管理</h2>
          <ul className="list-group admin-nav-link">
            <li className="list-group-item border-0">首页</li>
            <li className="list-group-item border-0">首页</li>
            <li className="list-group-item border-0">首页</li>
            <li className="list-group-item border-0">首页</li>
          </ul>
        </div>
      </div>
    )
  }

}

export default AdminNav
