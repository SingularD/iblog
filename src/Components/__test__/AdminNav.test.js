import React from 'react'
import { shallow } from "enzyme";
import AdminNav from '../Admin/AdminNav'

describe("检测后台管理页面", () => {
  let wrapper = shallow(<AdminNav />)
  it("匹配快照", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
