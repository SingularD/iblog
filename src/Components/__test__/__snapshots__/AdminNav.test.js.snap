// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`检测后台管理页面 匹配快照 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminNav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="col-10 pt-5"
      >
        <h2>
          后台管理
        </h2>
        <ul
          className="list-group admin-nav-link"
        >
          <li
            className="list-group-item border-0"
          >
            <Link
              to="/admin/"
            >
              首页
            </Link>
          </li>
          <li
            className="list-group-item border-0"
          >
            <Link
              to="/admin/writing"
            >
              写作
            </Link>
          </li>
          <li
            className="list-group-item border-0"
          >
            <Link
              to="/admin/edit"
            >
              编辑
            </Link>
          </li>
        </ul>
      </div>,
      "className": "admin-nav-container col-2 row justify-content-center bg-light",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            后台管理
          </h2>,
          <ul
            className="list-group admin-nav-link"
          >
            <li
              className="list-group-item border-0"
            >
              <Link
                to="/admin/"
              >
                首页
              </Link>
            </li>
            <li
              className="list-group-item border-0"
            >
              <Link
                to="/admin/writing"
              >
                写作
              </Link>
            </li>
            <li
              className="list-group-item border-0"
            >
              <Link
                to="/admin/edit"
              >
                编辑
              </Link>
            </li>
          </ul>,
        ],
        "className": "col-10 pt-5",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "后台管理",
          },
          "ref": null,
          "rendered": "后台管理",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/admin/"
                >
                  首页
                </Link>
              </li>,
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/admin/writing"
                >
                  写作
                </Link>
              </li>,
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/admin/edit"
                >
                  编辑
                </Link>
              </li>,
            ],
            "className": "list-group admin-nav-link",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  to="/admin/"
                >
                  首页
                </Link>,
                "className": "list-group-item border-0",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "首页",
                  "to": "/admin/",
                },
                "ref": null,
                "rendered": "首页",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  to="/admin/writing"
                >
                  写作
                </Link>,
                "className": "list-group-item border-0",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "写作",
                  "to": "/admin/writing",
                },
                "ref": null,
                "rendered": "写作",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  to="/admin/edit"
                >
                  编辑
                </Link>,
                "className": "list-group-item border-0",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "编辑",
                  "to": "/admin/edit",
                },
                "ref": null,
                "rendered": "编辑",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-10 pt-5"
        >
          <h2>
            后台管理
          </h2>
          <ul
            className="list-group admin-nav-link"
          >
            <li
              className="list-group-item border-0"
            >
              <Link
                to="/admin/"
              >
                首页
              </Link>
            </li>
            <li
              className="list-group-item border-0"
            >
              <Link
                to="/admin/writing"
              >
                写作
              </Link>
            </li>
            <li
              className="list-group-item border-0"
            >
              <Link
                to="/admin/edit"
              >
                编辑
              </Link>
            </li>
          </ul>
        </div>,
        "className": "admin-nav-container col-2 row justify-content-center bg-light",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              后台管理
            </h2>,
            <ul
              className="list-group admin-nav-link"
            >
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/admin/"
                >
                  首页
                </Link>
              </li>
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/admin/writing"
                >
                  写作
                </Link>
              </li>
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/admin/edit"
                >
                  编辑
                </Link>
              </li>
            </ul>,
          ],
          "className": "col-10 pt-5",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "后台管理",
            },
            "ref": null,
            "rendered": "后台管理",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/admin/"
                  >
                    首页
                  </Link>
                </li>,
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/admin/writing"
                  >
                    写作
                  </Link>
                </li>,
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/admin/edit"
                  >
                    编辑
                  </Link>
                </li>,
              ],
              "className": "list-group admin-nav-link",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    to="/admin/"
                  >
                    首页
                  </Link>,
                  "className": "list-group-item border-0",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "首页",
                    "to": "/admin/",
                  },
                  "ref": null,
                  "rendered": "首页",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    to="/admin/writing"
                  >
                    写作
                  </Link>,
                  "className": "list-group-item border-0",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "写作",
                    "to": "/admin/writing",
                  },
                  "ref": null,
                  "rendered": "写作",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    to="/admin/edit"
                  >
                    编辑
                  </Link>,
                  "className": "list-group-item border-0",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "编辑",
                    "to": "/admin/edit",
                  },
                  "ref": null,
                  "rendered": "编辑",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
